dialog.cpp

```diff
@@ -74,9 +74,9 @@ Dialog::Dialog(QWidget *parent) :
     setWindowTitle(QSL("LXQt Runner"));
     setAttribute(Qt::WA_TranslucentBackground);
 
-    connect(LXQt::Settings::globalSettings(), SIGNAL(iconThemeChanged()), this, SLOT(update()));
-    connect(ui->closeButton, SIGNAL(clicked()), this, SLOT(hide()));
-    connect(mSettings, SIGNAL(settingsChanged()), this, SLOT(applySettings()));
+    connect(LXQt::Settings::globalSettings(), &LXQt::GlobalSettings::iconThemeChanged, this, &Dialog::reload);
```

Comment:
Why defining `Dialog::reload()` below an using it here? We have `QWidget::update()` and it was correctly used here.

Reply:
> EDIT: I found out why you needed 

I still don't get it.  Why can't we simply use the `static_cast` or `qOverload`?

Reply:
> I still don't get it. Why can't we simply use the `static_cast` or `qOverload`?

Right! Sorry for the hack. I probably did something wrong while trying to implement it and I got convinced I had to workaround it.

Reply:
> I probably did something wrong...

IMO, your code wasn't wrong. My suggestion was about making it cleaner -- and @palinek's suggestion was still cleaner.